src/main/java/com/ericsson/ei/handlers/RmqHandler.java

```diff
@@ -150,6 +162,33 @@ public ConnectionFactory connectionFactory() {
         return cachingConnectionFactory;
     }
 
+    @Bean
+    public RabbitManagementTemplate deleteBindingKeys() {
+    	int rabbitPort=port;
+    	if(port==5672){
+    		rabbitPort=15672;
+    	}
+        rabbitManagementTemplate = new RabbitManagementTemplate("http://"+host+":"+rabbitPort+"/api/",user,password);
```

Comment:
RabbitManagementTemplate will use the http calls to connect with RabbitMQ and RabbitTemplate uses the amqp. both implementations are different. so better to change the new implementation to amqp based.

Reply:
As per raja's comment, we have investigated for the other way of getting the binding keys form the RabbitMq,but rather that RabbitManagmentTemplate/Client we couldn't find any way to fetch the list of bindings using current EI implementation (Spring AMQP). Also digged into the code level referring the RabbitTemplate, channels and connectionFactory but there is no such functionality to get the bindingKeys.

So other way of approach for this implementation is using files, i.e to store the binding keys into the file/List and check with the existing bindingkeys, add or update the file with the required binding keys on EI start. The file gets updated when the binding configuration is changed.With this approach the manually created bindings cannot be retrieved which in turn cannot be deleted from the code.
Please suggest if this can be implemented in any other way.

Reply:
How is other Eiffel component handling restart of Eiffel components with new bindingKey ?

Reply:
Can't EI create a new queue with the new BindingKey name as suffix to the queue name and let the administrator choose to do with old queue and old event data, if it should be kept or removed.
EI starts to consume messages from new queue with new binding key(s) when started with new binding keys.
By doing that we don't wipe any persistent data or what it can cause.

Reply:
Or if the administrator want the old queue and old bindingkeys to be removed automatically, set durable queue flag to False and old queue+bindingkeys will be removed automatically at shutdown and when EI starts a new queue will be created with new bindingkey(s), if I understand it correctly.

Reply:
> Or if the administrator want the old queue and old bindingkeys to be removed automatically, set durable queue flag to False and old queue+bindingkeys will be removed automatically at shutdown and when EI starts a new queue will be created with new bindingkey(s), if I understand it correctly.

yah, but apart from setting durable queue flag, the actually issue is still staying i.e old binding keys which are not in use are staying back in mb queue. So is there any other way programmatically to remove old binding keys in mb queue.

Reply:
Have you looked into Queue TTL:
https://www.rabbitmq.com/ttl.html#queue-ttl

So queue is removed by RabbitMq itself after expire time.
This involves adding a new rabbitmq property to application.properties.

Reply:
yah,but accordingly as far as I know, with the same queue there will be different binding keys, so  binding keys which are not in use must be removed, so remaining binding keys which are in use with that queue will stay. If TTL is used entire queue will be in effect irrespective of binding keys. so we are trying to get the binding keys from mb queue. Correct me if I am wrong.

Reply:
I think the only solution is to bind and unbind the queue when EI starts and shutsdowns.
Since EI is not aware of previous binding keys used in other/previous running EI instances.
Documentation
https://pika.readthedocs.io/en/latest/modules/channel.html?highlight=bind#pika.channel.Channel.queue_unbind

If I understand what you trying to achieve.

I thinks its not safe/secure to let next EI instance to figure out what previously binding key was used and remove it.

Other option is what I commented about in previous comment, create a new queue with the new binding key and let rabbitmq remove the old queue+bindingkey with Queue TTL flag.

Reply:
Had a discussion regarding the implementation way and the proposed solution suggested in the meeting is to use mongodb collection to store the binding keys and use the collection for removing the binding keys and updating the new binding keys.
Will start the implementation using mongo db with collection name "BindingKeys".

Reply:
If we see this behavior on latest version of Eiffel Intelligence (and I think it is the same) we will need to transfer this fix also to the master branch, which will be included in an upcoming 4.x release (when the API versioning has been added)

Reply:
Isn't "binding-keys" better collection name,, then its easier to administrate MongoDb if using mongo client command and typing the collection name in mongo shell.
Its easier to type only lower letters in the shell.

Reply:
This collection need to be unique also for each EI instance if same mongodb is used, so collection name need to be configurable from application.config.
So we get unique collection name per instance, if same MongoDb is used.
Same MongoDb is used by tests and also in Easy2Use, maybe in other setups as well.